### PR TITLE
feat: generic plan filename branch fallback to parent dir

### PR DESCRIPTION
**Pull request by bronislav**
## Summary

`ExtractBranchName` falls back to the parent directory's basename when the plan filename is generic (`tasks`, `plan`, `plans`, `index`, `readme` — case-insensitive). OpenSpec-style layouts like `changes/add-dark-mode/tasks.md` now yield `add-dark-mode` instead of `tasks`.

Directory date stripping uses a stricter regex (`dirDatePrefixRe`) — only genuine ISO/compact dates (`2024-01-15-`, `20260428-`) are stripped, so non-date numeric prefixes like `2fa-` or `404-` are left alone.

Fallback stays on the original filename when: path is bare/root, parent dir is also generic, or the stripped dir name is empty (e.g. pure-date dir `20260428`).

Key files: `pkg/plan/plan.go` — `ExtractBranchName`, `genericPlanFilenames`, `stripDirDatePrefix`

## Test plan

- [x] `make test` — 22 new table-driven cases (OpenSpec layouts, date stripping, edge cases)
- [x] `make lint`
- [x] no regressions in existing `ExtractBranchName` tests

### CODE REVIEW

File: docs/plans/20260428-generic-plan-filename-branch-fallback.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- Change is isolated to `pkg/plan/plan.go`; no callers change
+- No config, no flags — hardcoded generic-name set for now
+
+## Solution Overview
+
+1. Extract the date-strip logic from `ExtractBranchName` into a private `stripDatePrefix(name)` helper so it can be reused for both the filename and the fallback directory name.
+2. Add a package-level `genericPlanFilenames` set (map[string]bool) with entries: `tasks`, `plan`, `index`, `readme`. A brief comment explains the rationale.
+3. In `ExtractBranchName`, after computing the stripped filename, do a **case-insensitive** lookup:
+   - If generic: compute `dir := stripDatePrefix(filepath.Base(filepath.Dir(planFile)))`. Use `dir`
+     as the branch name unless `dir` is `.`, `/`, empty, or itself a generic name (nested case →
```
[the remark this answers is not in the document]
Comment (bronislav) [reply docs/plans/20260428-generic-plan-filename-branch-fallback.md]: Fixed

**Review by umputun — COMMENTED** [review]
stepping back from line-level findings, I want to question the approach itself.

**most of this is already solvable today, with zero code changes**

if the user creates the branch first via a 3-line wrapper (e.g. `git checkout -b "$(basename $(dirname $1))"; ralphex "$@"`), `preparePlanBranch` returns early with "already on feature branch" (`pkg/git/service.go:183`) and `CreateBranchForPlan` exits without ever calling `ExtractBranchName` (line 229-230). Branch derivation simply doesn't run. The OpenSpec UX gap on the common path is a wrapper script away from being solved.

the only path where the heuristic actually changes anything is `--worktree` mode, which calls `preparePlanBranch` with `requireDefault=true` (line 285) and uses `ExtractBranchName` for the worktree dir name (line 272). That's a narrower slice of users.

**why a `--branch` flag and not a heuristic?**

a `--branch=NAME` flag is strictly more general:
- works for OpenSpec, any other tool's layout, custom namespacing (`feat/foo`), capitalised variants, `spec.md`, `description.md`, etc
- ~5 lines, 1 test, no regex, no edge cases
- self-documenting in `--help`
- covers worktree mode too (the case the heuristic was actually needed for)

the heuristic only helps when filename is in the hardcoded set (`tasks`, `plan`, `plans`, `index`, `readme`) AND parent dir is meaningful AND not also in the set. A user with `spec.md` or `description.md` gets nothing. So it isn't actually generic, it's five-name-OpenSpec-shaped.

**why the heuristic approach is fragile, not just incomplete**

the implementation already has bugs that fall directly out of the heuristic-based approach:

- `pkg/plan/plan.go:28` - `dirDatePrefixRe`'s compact-date alternative `^\d{8}-?` strips *any* 8-digit prefix, not just genuine `YYYYMMDD`. So `openspec/changes/12345678-login/tasks.md` returns `login` and loses the issue ID. The PR description claims "non-date numeric prefixes are left alone" but only the test cases for `2fa-` / `404-` exercise that, and they pass because they aren't 8 digits. Real 8-digit dir names (timestamps, ticket IDs) get falsely treated as dates. Fixing this drives toward `time.Parse` validation or yet-tighter regex, which is the maintenance treadmill heuristics put you on.

- `pkg/plan/plan.go:183-184` - the godoc no longer describes the actual behaviour, and "generic" isn't defined anywhere a reader can reach without grepping for `genericPlanFilenames`. A flag-based design is self-documenting in `--help`; this one needs prose to explain when each branch fires.

these aren't items to fix in this PR; they're examples of the cost of growing the heuristic. A `--branch` flag has none of them by construction.
